examples/records/records.bal

```diff
@@ -1,23 +1,32 @@
 import ballerina/io;
 
-// This defines an open record type named `Student` which in addition to the fields defined here, allows additional
-// `string` fields.
-type Student record {|
+// This defines an open record type named `Student`. The `{` and `}` delimiters indicate that this record type
+// allows mapping values with additional pure-typed (i.e., `anydata|error`) fields in addition to the fields described
+// in the descriptor. The descriptor `record { }` is equivalent to `record {| (anydata|error)...; |}`.
+type Student record {
     string name;
     int age;
     Grades grades;
-    // This is a `string` typed rest field. All additional fields should be of the rest field's type or a subtype of it.
-    // In the absence of an explicit rest field, an implicit rest field of type `anydata|error` is assumed.
-    string...;
+};
+
+// This defines a closed record type named `Address`. The `{|` and `|}` delimiters indicate that this record type
+// exclusively allows mapping values which contain the described fields.
```

Comment:
By exclusively are we trying to say we only allow the fields specified? How about something like the following instead?
```suggestion
// allows mapping values which contain only the described fields.
```

IMO, "exclusively allows mapping values" sounds like only this record type allows mapping values.

Reply:
Yeah, meant that only mapping values with those fixed set of fields are allowed